### PR TITLE
Support CancellationToken without onComplete - core alpha 881

### PR DESCRIPTION
**Pull request by DavidBoike**

### CODE REVIEW

**Comment by DavidBoike** [comment]
The failing test is inexplicable and only happens on TeamCity. Hoping it just goes away when I re-run on Monday.

File: src/NServiceBus.Transport.SQS/InputQueuePump.cs [hunk 1]
```diff
@@ -66,14 +66,16 @@ public async Task Initialize(PushRuntimeSettings limitations, OnMessage onMessag
             this.onError = onError;
         }
 
-        public Task StartReceive(CancellationToken cancellationToken)
+        public Task StartReceive(CancellationToken cancellationToken = default)
         {
-            if (tokenSource != null)
+            if (messagePumpCancellationTokenSource != null)
```
Comment (andreasohlund) [inline src/NServiceBus.Transport.SQS/InputQueuePump.cs:71]: This is the only of our pumps that guard for multiple calls to start, should we add this to all the others?

File: src/NServiceBus.Transport.SQS/InputQueuePump.cs [hunk 2]
```diff
@@ -100,20 +102,21 @@ public Task StartReceive(CancellationToken cancellationToken)
...
             }
 
             return Task.CompletedTask;
         }
 
-        public async Task StopReceive(CancellationToken cancellationToken)
+        public async Task StopReceive(CancellationToken cancellationToken = default)
         {
-            if (tokenSource == null)
+            if (messagePumpCancellationTokenSource == null)
```
Comment (andreasohlund) [inline src/NServiceBus.Transport.SQS/InputQueuePump.cs:113]: This is the only transport that guards for multiple calls to stop, should we align the others?

**Review by andreasohlund — APPROVED** [review]

File: src/NServiceBus.Transport.SQS/InputQueuePump.cs (hunk 2 above)
Comment (DavidBoike) [reply src/NServiceBus.Transport.SQS/InputQueuePump.cs:113]: Core already makes stopping a stopped endpoint a no-op: https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Unicast/RunningEndpointInstance.cs#L25-L28

If we're going to spend cycles aligning anything, I'd like it to be in removing the duplicate code in the first place.

File: src/NServiceBus.Transport.SQS/InputQueuePump.cs (hunk 1 above)
Comment (DavidBoike) [reply src/NServiceBus.Transport.SQS/InputQueuePump.cs:71]: Same thing as for Stop: https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Hosting/HostCreator.cs#L75-L82

**Pull request merged**